web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx

```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
   );
 };
 
-// PORT_REGEXP only allows digits with length 4.
-export const PORT_REGEX = /^\d{4}$/;
-const requirePort = value => () => {
-  const isValidId = value.match(PORT_REGEX);
-  if (!isValidId) {
+// Only allows digits with valid port range 1-65535.
+const requirePort = (value: string) => () => {
+  const numberValue = parseInt(value);
+  const isValidPort =
+    Number.isInteger(numberValue) && numberValue >= 1 && numberValue <= 65535;
+  if (!isValidPort) {
     return {
       valid: false,
-      message: 'port must be 4 digits',
+      message: 'Endpoint Port must be a valid port (1 - 65535).',
```

Comment:
the port input box width is quite small, so this err message will probably cause things to wrap and jump, what do you think about these messages:

`invalid port (1-65535)` or
`invalid range (1-65535)` 

if none of these is clear enough we might need to increase the port width

Reply:
done, it fits! 
<img width="542" alt="image" src="https://user-images.githubusercontent.com/60704961/217935251-bb3f13c4-902f-411e-b381-a167c17d4e58.png">

Reply:
please approve if you are good @zmb3